ika2btc/processor.go

```diff
@@ -0,0 +1,132 @@
+package ika2btc
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/gonative-cc/relayer/bitcoin"
+	"github.com/gonative-cc/relayer/dal"
+	"github.com/rs/zerolog/log"
+)
+
+// Processor handles processing transactions from IKA to Bitcoin.
+type Processor struct {
+	BtcClient               bitcoin.Client
+	db                      *dal.DB
+	txConfirmationThreshold uint8
+}
+
+// NewProcessor creates a new Processor instance.
+func NewProcessor(
+	btcClientConfig rpcclient.ConnConfig,
+	confirmationThreshold uint8,
+	db *dal.DB,
+) (*Processor, error) {
+
+	if db == nil {
+		err := fmt.Errorf("database cannot be nil")
+		log.Err(err).Msg("")
+		return nil, err
```

Comment:
Error logging should be in the main package ideally. Debug / info should be "locally". Also, let's move the error construction to a  `const`.
```suggestion
		return nil, ErrNoDB
```

Reply:
And error in `errors.go`:
```go
const ErrNoDB = errors.New("database can not be nil")
```

Reply:
Done